tests/api-watch/test_diff_class_type_decl.ml

```diff
@@ -0,0 +1,95 @@
+open Api_watch
+open Types
+
+let%test_module "Class type declaration diffing" =
+  (module struct
+    let basic_class_type_decl =
+      {
+        clty_params = [];
+        clty_type =
+          Cty_signature
+            {
+              csig_self = Ctype.newvar ();
+              csig_self_row = Ctype.newvar ();
+              csig_vars = Vars.empty;
+              csig_meths = Meths.empty;
+            };
+        clty_path = Path.Pident (Ident.create_local "test_class");
+        clty_hash_type =
+          {
+            type_params = [];
+            type_arity = 0;
+            type_kind = Type_abstract Definition;
+            type_private = Public;
+            type_manifest = None;
+            type_variance = [];
+            type_separability = [];
+            type_is_newtype = false;
+            type_expansion_scope = Btype.lowest_level;
+            type_loc = Location.none;
+            type_attributes = [];
+            type_immediate = Unknown;
+            type_unboxed_default = false;
+            type_uid = Uid.internal_not_actually_unique;
+          };
+        clty_variance = [];
+        clty_loc = Location.none;
+        clty_attributes = [];
+        clty_uid = Uid.internal_not_actually_unique;
+      }
+
+    let signature_with_class_type decl =
+      [
+        Sig_class_type
+          (Ident.create_local "test_class", decl, Trec_first, Exported);
+      ]
```

Comment:
You can use our `Test_helpers` library instead to get the compiled interface from OCaml sources. 

The function you want here is `Test_helpers.compile_interface`.

You can take a look at some existing expect tests we have in `test_diff.ml` or `test_diff_module.ml` to see an example of how it's used.